### PR TITLE
Feature/coding standards

### PR DESCRIPTION
**Pull request by defunctl**
## What does this do/fix?

Moves the coding standard to an external repo so we can update it and use different versions for different projects: https://github.com/moderntribe/coding-standards.

The 1.0.x branch will remain a mostly formatting branch and 2.0.x will become stricter and focus on better quality code and not just formatting.

This will mostly benefit Vscode users as the formatting properly matches (for the most part) what our PHPStorm settings are.

* Updated: Move phpcs to https://github.com/moderntribe/coding-standards
* Removed: Broken Gallery shortcode/subscriber.
* Removed: phpcs.xml, this should be git ignored.
* Added: phpcs.xml.dist with new coding standards (mostly formatting based for now).
* Updated: Updated all relevant PHP files for phpcs.
* Updated: .editorconfig to better match new phpcs formatting.  
* Removed: mercator mu plugin https://github.com/humanmade/Mercator as multisite domain mapping has been in core for some time.
* Updated: workflows/phpcs.yml to use the new coding standard in GitHub workflows  
* Fixed: Multiple blocks keywords not using `__()` properly.
* Updated: Added mu-plugins to husky hooks phpcs git commit hook. 

## QA

This pretty much touched every file, but it's currently using lighter phpcs rules that focus more on formatting rather than proper code. 

Testing locally is working well.

## Tests

Does this have tests?

- [ ] Yes
- [X] No, this doesn't need tests because it's a code quality/tooling update.
- [ ] No, I need help figuring out how to write the tests.

### CODE REVIEW

File: wp-content/plugins/core/src/Assets/Admin/Scripts.php [hunk 1]
```diff
@@ -52,4 +44,5 @@ private function localize_scripts( string $handle ): void {
 		wp_localize_script( $handle, 'modern_tribe_admin_i18n', $this->localization->get_data() );
```
Comment (nickpelton) [inline wp-content/plugins/core/src/Assets/Admin/Scripts.php:44]: Ran acceptance tests locally and every test failed from this error:
```
TypeError: Argument 1 passed to Tribe\Project\Assets\Admin\Scripts::localize_scripts() must be of the type string, bool given, called in /application/www/wp-content/plugins/core/src/Assets/Admin/Scripts.php on line 40 in file /application/www/wp-content/plugins/core/src/Assets/Admin/Scripts.php on line 43
Stack trace:
  1. TypeError-&gt;() /application/www/wp-content/plugins/core/src/Assets/Admin/Scripts.php:43
  2. Tribe\Project\Assets\Admin\Scripts-&gt;localize_scripts() /application/www/wp-content/plugins/core/src/Assets/Admin/Scripts.php:40
  3. Tribe\Project\Assets\Admin\Scripts-&gt;enqueue_scripts() /application/www/wp-content/plugins/core/src/Assets/Assets_Subscriber.php:62
  4. Tribe\Project\Assets\Assets_Subscriber-&gt;Tribe\Project\Assets\{closure}() /application/www/wp/wp-includes/class-wp-hook.php:290
  5. WP_Hook-&gt;apply_filters() /application/www/wp/wp-includes/class-wp-hook.php:316
  6. WP_Hook-&gt;do_action() /application/www/wp/wp-includes/plugin.php:484
  7. do_action() /application/www/wp/wp-admin/admin-header.php:102
  8. require_once() /application/www/wp/wp-admin/term.php:73
 ```

**Review by nickpelton — CHANGES_REQUESTED** [review]
* Updated fine
* Required updating PHPStorm to point at the phpcs.xml.dist for updated ruleset
* Acceptance tests fail due to failed type check

**Comment by nickpelton** [comment]
We should do a deeper ruleset review on the next BE meeting. As some checks are still PHPStorm only and not a part of PHPCS.

File: wp-content/plugins/core/src/Assets/Admin/Scripts.php (hunk 1 above)
Comment (defunctl) [reply wp-content/plugins/core/src/Assets/Admin/Scripts.php:44]: You don't have the core theme active but the core plugin.

**Comment by defunctl** [comment]
> We should do a deeper ruleset review on the next BE meeting. As some checks are still PHPStorm only and not a part of PHPCS.

Sounds good, the 2.0.x branch has the more stricter standards: https://github.com/moderntribe/coding-standards/tree/2.0.x but would require much more manual intervention on our core files so we can do that later once things slow down.

File: wp-content/plugins/core/src/Assets/Admin/Scripts.php (hunk 1 above)
Comment (nickpelton) [reply wp-content/plugins/core/src/Assets/Admin/Scripts.php:44]: Cleared the DB same issue. I was also under the impression that it does a fresh import for every test and should properly activate the theme and plugins required for the test.

Comment (defunctl) [reply wp-content/plugins/core/src/Assets/Admin/Scripts.php:44]: It should, not sure if it activates the theme though. I'll fix the bad typing on this this though.

Comment (defunctl) [reply wp-content/plugins/core/src/Assets/Admin/Scripts.php:48]: Pull and give that a try

Comment (nickpelton) [reply wp-content/plugins/core/src/Assets/Admin/Scripts.php:48]: Passes after these updates and my own local config issue was fixed.

File: phpcs.xml.dist [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <!-- Directories to be checked -->
+    <file>./wp-content/plugins/core</file>
+    <file>./wp-content/themes/core</file>
+    <file>./wp-content/mu-plugins</file>
+
+    <!-- Exclude files -->
+    <exclude-pattern>*-config.php</exclude-pattern>
+    <exclude-pattern>*vendor/</exclude-pattern>
+    <exclude-pattern>*tests/*</exclude-pattern>
+    <exclude-pattern>*.twig</exclude-pattern>
```
Comment (nickpelton) [inline phpcs.xml.dist:28]: We don't use twig anymore, this can likely be removed.

**Review by nickpelton — APPROVED** [review]
Minor note on twig. 🚀

**Pull request merged**